client-linuxapp/src/serviceinfo.rs

```diff
@@ -131,36 +131,44 @@ fn perform_rhsm(organization_id: &str, activation_key: &str, perform_insights: b
 #[derive(Debug)]
 struct BinaryFileInProgress {
     path: Option<String>,
+    prefix: Option<String>,
     length: Option<u64>,
     contents: Option<Vec<u8>>,
     mode: Option<u32>,
     digest: Option<Hash>,
 }
 
 impl BinaryFileInProgress {
-    fn new() -> Self {
+    fn new(prefix: Option<String>) -> Self {
         BinaryFileInProgress {
             path: None,
+            prefix,
             length: None,
             contents: None,
             mode: None,
             digest: None,
         }
     }
 
-    fn deploy(self) -> Result<()> {
-        let path = self.path.as_ref().unwrap();
-
-        let path = if let Ok(val) = env::var("BINARYFILE_PATH_PREFIX") {
-            PathBuf::from(&val).join(path)
-        } else {
-            PathBuf::from(path)
-        };
+    fn destination_path(path_str: &String, prefix: Option<String>) -> Result<PathBuf> {
+        let path = PathBuf::from(path_str);
 
         if !path.is_absolute() {
             bail!("Binary file path must be absolute");
         }
 
+        if let Some(val) = prefix {
+            // make sure we drop the first slash or PathBuf returns the join'ed absolute path
+            Ok(PathBuf::from(&val).join(&path_str[1..path_str.len()]))
```

Comment:
Fun thing: this assumes POSIX, i.e. non-Windows.

Reply:
don't we assume client-linuxapp only runs on linux which should mostly be posix?

Reply:
I have tried to ensure we don't always assume that. I don't see a reason why it couldn't run on Windows.

Reply:
fixed, as per your suggestion, I've added a conditional compilation and error out on !unix